src/psyclone/psyGen.py

```diff
@@ -4705,6 +4968,300 @@ def _number_handler(self, node, parent):
+    :raises NotImplementedError: Provided parameters are not supported yet.
+    :raises TypeError: Provided parameters have invalid error type.
+    :raises ValueError: Provided parameters contain invalid values.
+    '''
+
+    # Tuple with the valid values for the access attribute.
+    valid_scope_types = ('local', 'global_argument')
+    # Tuple with the valid datatypes.
+    valid_data_types = ('real', 'integer', 'character')
+
+    def __init__(self, name, datatype, shape=[], scope='local',
+                 is_input=False, is_output=False):
+
+        self._name = name
+
+        if datatype not in Symbol.valid_data_types:
+            raise NotImplementedError(
+                "Symbol can only be initialized with {0} datatypes."
+                "".format(str(Symbol.valid_data_types)))
+        self._datatype = datatype
+
+        if not isinstance(shape, list):
+            raise TypeError("Symbol shape attribute must be a list.")
+
+        if False in [isinstance(x, (type(None), int)) for x in shape]:
+            raise TypeError("Symbol shape list elements can only be "
+                            "'integer' or 'None'.")
+        self._shape = shape
+
+        # The following attributes have setter methods (with error checking)
+        self.scope = scope
+        self.is_input = is_input
+        self.is_output = is_output
+
+    @property
+    def name(self):
+        '''
+        :return: Name of the Symbol.
+        :rtype: string
+        '''
+        return self._name
+
+    @property
+    def datatype(self):
+        '''
+        :return: Datatype of the Symbol.
+        :rtype: string
+        '''
+        return self._datatype
+
+    @property
+    def is_input(self):
+        '''
+        :return: Whether the data already exists before kernel entry.
+        :rtype: bool
+        '''
+        return self._is_input
+
+    @is_input.setter
+    def is_input(self, new_is_input):
+        '''
+        :param bool new_is_input: Whether it is an already existent data that \
+                                  is  passed into the kernel upon entry.
+        :raises TypeError: Provided parameters have invalid error type.
+        :raises ValueError: 'new_is_input' contains an invalid value.
+        '''
+        if not isinstance(new_is_input, bool):
+            raise TypeError("Symbol 'is_input' attribute must be a boolean.")
+        if self.scope == 'local' and new_is_input is True:
+            raise ValueError("Symbol with 'local' scope can not have "
+                             "'is_input' attribute set to True.")
+        self._is_input = new_is_input
+
+    @property
+    def is_output(self):
+        '''
+        :return: Whether the data survives after kernel exit.
+        :rtype: bool
+        '''
+        return self._is_output
+
+    @is_output.setter
+    def is_output(self, new_is_output):
+        '''
+        :param bool new_is_output: Whether it is survives outside the kernel \
```

Comment:
"Whether the variable represented by this symbol survives outside..."

Reply:
Fixed